reference/supervisor.md

```diff
@@ -767,9 +745,40 @@ indicates the success of the save operation. Be aware that this function can
 overwrite silently existing files, so that the corresponding data may be lost.
 
 > **Note** [C++, Java, Python, Matlab]:
-In the other APIs, the `Robot.saveWorld()` function can be called without
+In the other APIs, the `Robot.worldSave()` function can be called without
 argument. In this case, a simple save operation is performed.
 
+The `wb_supervisor_world_reload()` function sends a request to the simulator
+process, asking it to reload the current world immediately. As a result of
+reloading the current world, the supervisor process and all the robot processes
+are terminated and restarted. You may wish to save some data in a file from your
+supervisor program in order to reload it when the supervisor controller
+restarts.
+
+---
+
+**Name**
+
+**wb\_supervisor\_simulation\_reset** - *reset the simulation*
+
+{[C++](cpp-api.md#cpp_supervisor)}, {[Java](java-api.md#java_supervisor)}, {[Python](python-api.md#python_supervisor)}, {[Matlab](matlab-api.md#matlab_supervisor)}, {[ROS](ros-api.md)}
+
+```c
+#include <webots/supervisor.h>
+
+void wb_supervisor_simulation_reset();
+```
+
+**Description**
+
+The `wb_supervisor_simulation_reset()` function sends a request to the simulator process, asking it to reset the simulation immediately.
+The simulation is not entirely destructed and reconstructed like when calling `wb_supervisor_world_reload`, but the initial state of all the nodes is restored, which is way faster.
+This function resets also the seed of the random number generator used in Webots, so that noise-based simulations can be be reproduced identically after calling this function.
+There are currently some limitations to the reset, The nodes removed from the simulation (either manually or from a Supervisor) are not re-imported during the reset.
+Furthermore, changes to node fields done manually or using a Supervisor are not reverted.
```

Comment:
`changes to node fields done manually or using a Supervisor`: so which changes are reverted?

Reply:
You should better described which changes are reverted and which are not, because it is still not clear.

Reply:
The addition of the `in the scene tree` helps. about the Supervisor part it was already clear before.
But if I would read this sentence for the first time the question I will have is always the same: if all the changes I apply are not reverted which changes are reverted?

I think that there should be a generic rule to define which fields are reverted, otherwise if each field behaves different it doesn't seem user-friendly.

Reply:
Just to summarize and help find a better explanation/generic rule, here is what is happening step by step when doing a simulation reset:
  - Simulation time is set to 0.0
  - All the nodes added during the simulation are removed.
  - All the sound sources are stoped.
  - The random seeds are reset.
  - All the nodes are reset:
    - **Brake**: The brake is released.
    - **Charger**: The battery level is restored (the `battery` **field** is restored), `emissiveColor` *field* of Material is restored.
    - **Connector**: If attached the connector is detached and the value of the `isLocked` **field** is restored.
    - **Emitter-Receiver**: the message queue is cleared.
    - **Joint/Motor**: The position/velocity/torque/force/acceleration is restored.
    - **LED**: The light is switched off. The emissiveColor **field** of the Material is restored.
    - **Lidar**: The position of the rotating head is restored.
    - **Pen**: All the painted texture are cleaned.
    - **Propeller**: The slow helix and it's initial position are restored.
    - **Robot**: The battery level is restored (`battery` **field** is restored). The controller is restarted.
    - **Solid**: The initial translation and rotation are restored (`translation` and `rotation` **fields** are restored) and the physic is reset.
    - **Supervisor**: All the labels are removed.
    - **Track**: the motor position is restored and the translation of the textureTransform is restored (`translation` **field** is restored).
    - **Viewpoint**: the position and orientation **fields** are restored.

The only generic rule I see is: the restored fields are the one changed by the simulation itself and not by an external intervention (manually in the scene tree or by a Supervisor).

Reply:
The generic rule that I see is that the devices and objects internal states modified by the simulation itself and by the controller are reset.

Reply:
But it is very important to write this as clearly as possible in the documentation.

Reply:
Examples are also useful to explain the behavior, you do not have to list everything, but probably mentioning the most important things that are reset will help the user.